metrics/sample.go

```diff
@@ -179,7 +180,7 @@ func parsePercentile(stat string) (float64, error) {
 // All methods should not panic, even if they are called on a nil pointer.
 //easyjson:skip
 type SampleTags struct {
-	tags map[string]string
+	n    *atlas.Node
```

Comment:
I would recommend to keep this attribute named `tags` for clarity. I personally find it hard to follow code when anything other than a local variable is named with a single letter. Adding a documentation comment could be useful to explicit what to expect from an `*atlas.Node` too?